include/nn/sl/CacheManager.h

```diff
@@ -0,0 +1,95 @@
+#pragma once
+
+#include <nn/result.h>
+#include <nn/sl/ISerializer.h>
+#include <nn/sl/KillerNotification.h>
+#include <nn/sl/sl_cpp.h>
+#include <wut.h>
+
+#ifdef __cplusplus
+
+namespace nn::sl {
```

Comment:
Nested namespaces are only available since C++17. As long as there isn't anything else which requires a newer C++ standard, it probably makes sense to avoid this.

Reply:
gcc doesn't give me a warning/error when using `-std=c++11` and `-Wall` so it might be fine to use it anyway?